File: app11_swift_ios/SwiftGuardData/Sources/SwiftGuardData/Integrity/Hashing.swift
import CryptoKit
import Foundation

/// Thin wrapper over Apple's own `CryptoKit` — no third-party crypto
/// dependency needed for SHA-256, unlike every server-based sibling that had
/// to pick a library for this.
enum Hashing {
    static func sha256Hex(_ input: String) -> String {
        sha256Hex(Data(input.utf8))
    }

    static func sha256Hex(_ data: Data) -> String {
        SHA256.hash(data: data).map { String(format: "%02x", $0) }.joined()
    }
}

File: app12_kotlin_android/data/src/main/kotlin/com/kotlinguard/data/integrity/Hashing.kt
package com.kotlinguard.data.integrity

import java.security.MessageDigest

/**
 * `java.security.MessageDigest` is part of the JDK/Android platform — no
 * third-party crypto dependency needed for SHA-256, the same "free" property
 * app11 gets from Apple's `CryptoKit`, and every server-based sibling had to
 * pick a library for.
 */
object Hashing {
    fun sha256Hex(input: String): String = sha256Hex(input.toByteArray(Charsets.UTF_8))

    fun sha256Hex(data: ByteArray): String =
        MessageDigest.getInstance("SHA-256").digest(data).joinToString("") { "%02x".format(it) }
}
